### PR TITLE
Add OSS support

### PR DESCRIPTION
**Pull request by vincent-karuri**
Adds support for saving multimedia to OSS.

Depends on https://github.com/OpenSRP/opensrp-server-core/pull/214.

### CODE REVIEW

File: src/main/java/org/opensrp/web/config/security/BasicAuthSecurityConfig.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected void configure(HttpSecurity http) throws Exception {
 		.mvcMatchers("/**").hasRole(Role.OPENMRS)
 		.and()
     		.csrf()
-    		.ignoringAntMatchers("/rest/**","/location/**");
+    		.ignoringAntMatchers("/rest/**","/location/**", "/multimedia/upload");
```
Comment (githengi) [inline src/main/java/org/opensrp/web/config/security/BasicAuthSecurityConfig.java:66]: This is  so as multimedia endpoints use custom headers to authenticate. Has the client side been modified to support normal authentication

File: src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -62,10 +62,10 @@
         <filter-class>org.opensrp.web.GzipBodyDecompressFilter</filter-class>
     </filter>
 
-    <filter>
+   <!-- <filter>
         <filter-name>CrossSiteScriptingPreventionFilter</filter-name>
         <filter-class>org.opensrp.web.config.security.filter.CrossSiteScriptingPreventionFilter</filter-class>
-    </filter>
+    </filter> -->
```
Comment (githengi) [inline src/main/webapp/WEB-INF/web.xml:68]: This has been fixed, remove this

File: src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -97,12 +97,12 @@
...
 	</filter-mapping>
 
-    <filter-mapping>
-        <filter-name>CrossSiteScriptingPreventionFilter</filter-name>
-        <url-pattern>/*</url-pattern>
-    </filter-mapping>
+    <!-- <filter-mapping>
+		 <filter-name>CrossSiteScriptingPreventionFilter</filter-name>
+		 <url-pattern>/*</url-pattern>
+	 </filter-mapping> -->
```
Comment (githengi) [inline src/main/webapp/WEB-INF/web.xml:103]: This has been fixed, remove this

File: src/main/java/org/opensrp/web/config/security/BasicAuthSecurityConfig.java (hunk 1 above)
Comment (vincent-karuri) [reply src/main/java/org/opensrp/web/config/security/BasicAuthSecurityConfig.java:66]: Not sure I understand. Could you rephrase this?

I did not have to make any changes on the client.

File: src/main/webapp/WEB-INF/web.xml (hunk 3 above)
Comment (vincent-karuri) [reply src/main/webapp/WEB-INF/web.xml:103]: I removed this filter because multimedia was being corrupted.

Is this the fix you're referring to?

File: src/main/webapp/WEB-INF/web.xml (hunk 2 above)
Comment (vincent-karuri) [reply src/main/webapp/WEB-INF/web.xml]: Done.

File: src/main/webapp/WEB-INF/web.xml (hunk 3 above)
Comment (vincent-karuri) [reply src/main/webapp/WEB-INF/web.xml]: Done.

File: src/main/java/org/opensrp/web/rest/PlanResource.java [hunk 4]
```diff
@@ -217,10 +216,10 @@ public void setLocationService(PhysicalLocationService locationService) {
 	 */
 	@RequestMapping(value = "/findLocationNames/{planIdentifier}", method = RequestMethod.GET, produces = {
 	        MediaType.APPLICATION_JSON_VALUE })
-	public ResponseEntity<List<LocationDetail>> findLocationDetailsByPlanId(
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/PlanResource.java:220]: Why is this being changed on this PR

Comment (vincent-karuri) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: A test was failing - not sure how it passed on the main branch. 

The issue here is that the test expects null fields to be ignored but Spring's serialization doesn't ignore them - so I modified it to use gson.

Comment (githengi) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: I think the main reason for now using domain objects instead  of string is also to have schema documentation in swagger. Why not change the test instead of changing implementation here

Comment (vincent-karuri) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: Most of the other methods use gson serialization in this class. Also, the test is probably a reflection of what the code was supposed to do, so modifying that instead of the code did not feel right.

Comment (githengi) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: The test is passing in v2.1 and master branches

Comment (vincent-karuri) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: This https://travis-ci.org/github/OpenSRP/opensrp-server-web/builds/693776168 shows it was failing after the merge.

Comment (vincent-karuri) [reply src/main/java/org/opensrp/web/rest/PlanResource.java:220]: It was also failing locally.
